x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -268,21 +268,31 @@ export class CloudPlugin implements Plugin<CloudSetup> {
               user.authentication_realm?.name === 'cloud-saml-kibana'
             ) {
               // If authenticated via Cloud SAML, use the SAML username as the user ID
-              return user.username;
+              return { userId: user.username, isElasticCloudUser: true };
             }
 
-            return cloudId ? `${cloudId}:${user.username}` : user.username;
+            return {
+              // Join the cloud org id and the user to create a truly unique user id.
```

Comment:
I think we need to explain why we're doing things differently in the two cases (compatibility reasons, mostly)

Reply:
That's a good point! I extended the comment to explain it further :)
Please, let me know if it needs more context.